### ISSUE TITLE
Canceling secret creation in kv v2 leads to broken UI state

### ISSUE THREAD

**Issue by Yorwba** [hashicorp/vault#20983]
**To Reproduce**
1. Visit `/ui/vault/secrets/<engine>/list` in a browser, where `<engine>` uses v2 of the `kv` backend.
2. Click "Create secret +"
3. Click "Cancel"
4. A warning appears that navigating away will discard unchanged changes. Click "OK"
5. The list of secrets is now empty, and the "Filter secrets" input and "Create secret +" button are gone. Refreshing the page is required to make the UI usable again.

**Expected behavior**
Canceling the secret creation should return the UI to the state it was in before, or at least something non-broken.

**Environment:**
* Vault Server Version (retrieve with `vault status`): 1.13.2
* Vault CLI Version (retrieve with `vault version`): 1.13.2
* Browser: Firefox 102.11.0esr

**Additional context**
This problem does not appear with the `cubbyhole` engine.

**Comment by mpalmi** [comment]
I've closed the duplicate (#21303), but adding in the additional context provided in that Issue:

> Additional context
> 
> When this happens the following JS error is visible in the browser console:
>
```
vendor-7da855e798cb22554cdbb9aa84703a8f.js:8386 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'charAt')
    at e.default (vendor-7da855e798cb22554cdbb9aa84703a8f.js:8386:27)
    at e.<anonymous> (vendor-7da855e798cb22554cdbb9aa84703a8f.js:8169:22)
    at vendor-7da855e798cb22554cdbb9aa84703a8f.js:696:67
    at e.untrack (vendor-7da855e798cb22554cdbb9aa84703a8f.js:2968:12)
    at me.get (vendor-7da855e798cb22554cdbb9aa84703a8f.js:696:56)
    at e.r [as firstPartialMatch] (vendor-7da855e798cb22554cdbb9aa84703a8f.js:672:245)
    at je (vendor-7da855e798cb22554cdbb9aa84703a8f.js:715:73)
    at vendor-7da855e798cb22554cdbb9aa84703a8f.js:2314:39
    at vendor-7da855e798cb22554cdbb9aa84703a8f.js:2309:41
    at e.track (vendor-7da855e798cb22554cdbb9aa84703a8f.js:2965:5)
```
>
> With that null I assume it's trying to redraw the page before it's made a request for the data to do so.

**Comment by hashishaw** [comment]
Hi @Yorwba, thank you for opening this issue and for the detailed replication steps! We have fixed this in #22541 which was included in the 1.13.6 release.
